swagger_server/controllers_local/collectioninstrument.py

```diff
@@ -0,0 +1,300 @@
+                used.append(ExerciseModel)
+            elif attr == 'survey' and SurveyModel not in used:
+                query = query.join(SurveyModel.instruments)
+                used.append(SurveyModel)
+            elif attr in classifications and ClassificationModel not in used:
+                query = query.join(ClassificationModel.instrument)
+                used.append(ClassificationModel)
+            else:
+                return 'unable to handle query for ({}={})'.format(attr, value)
+        for attr, value in classifiers.items():
+            if attr == 'ru_ref':
+                query = query.filter(BusinessModel.ru_ref == value)
+            elif attr == 'exercise':
+                query = query.filter(ExerciseModel.exercise_id == UUID(value))
+            elif attr == 'survey':
+                query = query.filter(SurveyModel.survey_id == UUID(value))
+            else:
+                query = query.filter(ClassificationModel.kind == attr).filter(ClassificationModel.value == value)
+        return query.all()
+
+    @protect(uuid=True)
+    def define_batch(self, exercise_id, count):
+        """
+        Define a batch is used to tell the service which surveys are in play and how many files to expect.
+        At some point this will be connected to a user to a UI which will allow entry of these details, but
+        for testing batches can be initiated with the Swagger UI
+        
+        :param exercise_id: The survey id (this is a uuid)
+        :param count: The number of files we are expecting (in total)
+        :return: At present this routine can't fail as we have no spec for parameter limits
+        """
+        exercise = self._get_exercise(exercise_id)
+        if not exercise:
+            ons_env.session.add(ExerciseModel(exercise_id=exercise_id, items=count))
+            ons_env.session.commit()
+        return 200, {'text': 'OK'}
+
+    @protect(uuid=True)
+    def clear(self, exercise_id):
+        """
+        Clear a batch (or reject a batch) is used to scrap a current batch in the event the user decides they
+        don't actually want to commit it for some reason.
+
+        :param exercise_id: An exercise id, which is a uuid
+        :return: Returns a 200 if the batch was removed, or 204 if not found
+        """
+        exercise = self._get_exercise(exercise_id)
+        if not exercise:
+            return 204, {'text': 'No such batch'}
+        ons_env.session.delete(exercise)
+        ons_env.session.commit()
+        return 200, {'text': 'OK'}
+
+    @protect(uuid=True)
+    def instrument(self, instrument_id):
+        """
+        Return a specific collection instrument based on an id
+
+        :param instrument_id: The id of the instrument we want 
+        :return: formatted JSON version of the instrument
+        """
+        instrument = self._get_instrument(instrument_id)
+        if not instrument:
+            return 404, 'Instrument not found'
```

Comment:
Good question - I don't believe we have an authoratative answer, the 204 was initially suggested by Duncan ..